sqlglot/parser.py

```diff
@@ -3864,7 +3911,18 @@ def _parse_merge(self) -> exp.Expression:
         )
 
     def _parse_set(self) -> exp.Expression:
-        return self.expression(exp.Set, expressions=self._parse_csv(self._parse_set_item))
+        index = self._index
+        try:
+            return self.expression(exp.Set, expressions=self._parse_csv(self._parse_set_item))
+        except ParseError:
```

Comment:
This could lead to bugs that are hard to debug, as we silence the error.

Reply:
Agreed, I don't feel very good about this too. Maybe we should get rid of `_parse_set_item` and `_default_parse_set_Item` and have `_parse_set_item_assignment` return `None` on failure, so we can gracefully handle it. How does that sound?

Reply:
you should definitely not use try catch . if there are any tokens left after a set, it’s failed

Reply:
> if there are any tokens left after a set, it’s failed

Not necessarily, since spark [supports](https://spark.apache.org/docs/3.0.0-preview/sql-ref-syntax-aux-conf-mgmt-set.html#examples) `SET;`. Agreed with the comments regarding try-catch though, I will try to remove that.

Reply:
if you parse set; there are no tokens left, just check for self._curr and there are none so you don’t raise

Reply:
Ignore me, I misread your comment and got confused, you're right.

Reply:
Yep, I realized what you mean now -- thanks for the suggestion

Reply:
Got rid of try / catch.